### ISSUE TITLE
irm exits success when it should exit with HIERARCHY_ERROR

### ISSUE THREAD

**Issue by keithj** [irods/irods#3566]
iRODS 4.1.10, icommands 4.1.10

The following illustrates the situation. The data object is on a resource that is unavailable. `iget` fails as expected, but `irm` reports success.

````
$ irm /seq/trash/orphan/file3
$ echo $?
0
$ ils /seq/trash/orphan/file3
  /seq/trash/orphan/file3

$ iget /seq/trash/orphan/file3 -
ERROR: getUtil: get error for ./file3 status = -1803000 HIERARCHY_ERROR
````

**Comment by tempoz** [comment]
I tried to reproduce this on 4.1.10 with 4.1.10 icommands. If I mark the resource down, I get a `HIERARCHY ERROR` for both `iget` and `irm`. If I actually take the resource down by shutting down the resource server hosting it, I get a `USER_SOCK_CONNECT_ERR` for both the `iget` and `irm`. I did this with the trash both on and off and got the same result. What did you do to make the resource 'unavailable'?
